service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/ReviewCMOControllerAboutToSubmitTest.java

```diff
@@ -117,22 +119,33 @@ void shouldSealPDFAndAddToSealedCMOsListWhenJudgeApprovesOrder() throws Exceptio
     }
 
     @Test
-    void shouldUpdateStateToIssueResolutionWhenNextHearingIssueResolutionAndCmoDecisionIsSendToAllParties() {
+    void shouldUpdateStateToIssueResolutionWhenNextHearingIsOfTypeIssueResolutionAndCmoDecisionIsSendToAllParties() {
         given(featureToggleService.isNewCaseStateModelEnabled()).willReturn(true);
 
         UUID cmoId = UUID.randomUUID();
-        CaseData caseData = buildCaseData(cmoId, SEND_TO_ALL_PARTIES);
+        CaseData caseData = buildCaseData(cmoId, SEND_TO_ALL_PARTIES, ISSUE_RESOLUTION);
         CaseData responseData = extractCaseData(postAboutToSubmitEvent(caseData));
 
         assertThat(State.ISSUE_RESOLUTION).isEqualTo(responseData.getState());
     }
 
+    @Test
+    void shouldUpdateStateToFinalHearingWhenNextHearingIssueIsOfTypeFinalAndCmoDecisionIsSendToAllParties() {
+        given(featureToggleService.isNewCaseStateModelEnabled()).willReturn(true);
```

Comment:
thoughts on another test shouldNotUpdateStateToFinalHearingWhenNextHearingIssueIsNotOfTypeFinal ? happy if you think it's covered enough though!

Reply:
I think we can maybe combine the existing.